apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/QueryOptions.cs

```diff
@@ -196,6 +196,19 @@ internal void ModifyRequest(JobConfigurationQuery query)
             {
                 query.TimePartitioning = TimePartitioning;
             }
+            if (SessionID != null)
```

Comment:
I'd like to avoid the duplication here.
I'd suggest add a protected method in JobCreationOptions of something like "GetConnectionProperties".

Reply:
done